gtfu/src/main/java/gtfu/tools/TripListGenerator.java

```diff
@@ -268,24 +285,16 @@ public static void main(String[] arg) throws Exception {
         tf.dispose();
         //Debug.log("- smap: " + smap);
 
-        for (int i=0; i<list.size(); i++) {
-            TripListEntry e = list.get(i);
-            String routeID = e.routeID;
-            String aid = rmap.get(e.routeID).agencyID;
-
-            if (gtfsAgency != null && !aid.equals(gtfsAgency)) {
-               list.remove(i--);
-            }
-        }
-
-        out.println();
+        // It's temping to replace the below manual JSON generation with JSONObject and JSONArray classes,
+        // but because they offer little control over formatting, this method is preferable for something
+        // needing to be read by humans.
```

Comment:
fasterxml's ObjectMapper has a method called writeValue() or something. While that doesn't support pretty-printing, you can get a pretty printer from the mapper and then call writeValue() on that. Not saying you have to, but you can if you want.